picasso/src/main/java/com/squareup/picasso3/Picasso.java

```diff
@@ -856,4 +858,56 @@ public Picasso build() {
       this.debugColor = debugColor;
     }
   }
+
+  @Override public void cacheMaxSize(int maxSize) {
+    for (EventListener listener : eventListeners) {
```

Comment:
Use an indexed loop for all of these to avoid allocating an iterator

Reply:
updated